src/input/cache/Manager.cxx

```diff
@@ -81,11 +81,9 @@ InputCacheManager::Get(const char *uri, bool create)
 	if (!PathTraitsUTF8::IsAbsolute(uri))
 		return {};
 
-	UriMap::insert_commit_data hint;
-	auto result = items_by_uri.insert_check(uri, items_by_uri.key_comp(),
-						hint);
-	if (!result.second) {
-		auto &item = *result.first;
+	auto iter = items_by_uri.find(uri, items_by_uri.key_comp());
+	if (iter != items_by_uri.end()) {
+		auto &item = *iter;
```

Comment:
I am not an experienced C++ programmer. This patch is the result of me bashing the keyboard until the build succeeds.

Reply:
I'm also using this binary to listen to music, with the input cache enabled. But I was never able to trigger the bug this should solve, so I don't consider that very thorough testing.